##
#   File: /patches/020-Notice-Access-Control-Allow-Origin-only-once.patch
#   Project: rclone
#   File Created: Sunday, 3rd May 2020 1:44:22 pm
#   Author: ElonH[EH](elonhhuang@gmail.com)
#   License: GNU General Public License v3.0 or later(http://www.gnu.org/licenses/gpl-3.0-standalone.html)
#   Copyright (C) 2020 [ElonH]
##
diff --git a/fs/rc/rcserver/rcserver.go b/fs/rc/rcserver/rcserver.go
index 4a2b5c71e..f00994e93 100644
--- a/fs/rc/rcserver/rcserver.go
+++ b/fs/rc/rcserver/rcserver.go
@@ -14,6 +14,7 @@ import (
 	"regexp"
 	"sort"
 	"strings"
+	"sync"
 
 	"github.com/pkg/errors"
 	"github.com/rclone/rclone/cmd/serve/httplib"
@@ -29,6 +30,8 @@ import (
 	"github.com/skratchdot/open-golang/open"
 )
 
+var onlyOnce sync.Once
+
 // Start the remote control server if configured
 //
 // If the server wasn't configured the *Server returned may be nil
@@ -172,9 +175,11 @@ func (s *Server) handler(w http.ResponseWriter, r *http.Request) {
 
 	allowOrigin := rcflags.Opt.AccessControlAllowOrigin
 	if allowOrigin != "" {
-		if allowOrigin == "*" {
-			fs.Logf(nil, "Warning: Allow origin set to *. This can cause serious security problems.")
-		}
+		onlyOnce.Do(func() {
+			if allowOrigin == "*" {
+				fs.Logf(nil, "Warning: Allow origin set to *. This can cause serious security problems.")
+			}
+		})
 		w.Header().Add("Access-Control-Allow-Origin", allowOrigin)
 	} else {
 		w.Header().Add("Access-Control-Allow-Origin", s.URL())
